prov/efa/src/efa_fabric.c

```diff
@@ -479,11 +491,39 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, MAX_GDR_SUPPORT_STRLEN);
+	if (ret < 0) {
+		if (errno == ENOENT) {
+			/* sysfs file does not exist, gdr is not supported */
+			ret = 0;
+		}
+
 		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	gdr_support_str[MAX_GDR_SUPPORT_STRLEN-1] = '\0';
+	ret = strtol(gdr_support_str, &endptr, 10);
+	if ((endptr - gdr_support_str) != 1) {
+		EFA_WARN(FI_LOG_FABRIC, "Unexpected sysfs file content! gdr_support_str: %s\n",
+			 gdr_support_str);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	if (ret != 0 && ret != 1) {
```

Comment:
Are we convinced that 1 is the only positive value we'll ever have?

Reply:
According to the driver source code 

https://github.com/amzn/amzn-drivers/blob/50aed2a9741efd9e854a0d4c345e90aeb3566ab2/kernel/linux/efa/src/efa_sysfs.c#L16

1 is the only positive value.

Reply:
That is not the question I asked.